src/Microsoft.AspNet.Mvc.Core/Formatters/OutputFormatterContext.cs

```diff
@@ -0,0 +1,52 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using Microsoft.AspNet.Http;
+using Microsoft.Net.Http.Headers;
+
+namespace Microsoft.AspNet.Mvc.Formatters
+{
+    /// <summary>
+    /// A mutable context object for use with <see cref="IOutputFormatter"/>.
+    /// </summary>
+    public class OutputFormatterContext : OutputFormatterWriteContext
+    {
+        /// <summary>
+        /// Gets or sets a value inidicating that content-negotiation could not find a formatter based on the 
+        /// information on the <see cref="Http.HttpRequest"/>.
+        /// </summary>
+        public new bool? FailedContentNegotiation
+        {
+            get { return base.FailedContentNegotiation; }
+            set { base.FailedContentNegotiation = value; }
+        }
+
+        /// <summary>
+        /// Gets or sets the <see cref="HttpContext"/> context associated with the current operation.
+        /// </summary>
+        public new HttpContext HttpContext
+        {
+            get { return base.HttpContext; }
+            set { base.HttpContext = value; }
+        }
+
+        /// <summary>
+        /// Gets or sets the object to write to the response.
+        /// </summary>
+        public new object Object
+        {
+            get { return base.Object; }
+            set { base.Object = value; }
+        }
+
+        /// <summary>
+        /// Gets or sets the <see cref="Type"/> of the object to write to the response.
+        /// </summary>
+        public new Type ObjectType
```

Comment:
Could you avoid `new` here if the properties in the `abstract` classes were `abstract` instead of `virtual`?